analyzer/src/main/kotlin/managers/utils/NpmDependencyHandler.kt

```diff
@@ -49,13 +49,13 @@ data class NpmModuleInfo(
 /**
  * A specialized [DependencyHandler] implementation for NPM.
  */
-class NpmDependencyHandler : DependencyHandler<NpmModuleInfo> {
+class NpmDependencyHandler(private val npm: Npm) : DependencyHandler<NpmModuleInfo> {
```

Comment:
This bidirectional dependency between `Npm` and `NpmDependencyHandler` is a bit problematic IMHO. Is there a way to move functionality, so that the relation becomes unidirectional again? For instance, could the `parsePackage()` function (and maybe further helper functions) be moved to `NpmDependencyHandler`?

Reply:
Unfortunately, `parsePackage()` needs to call a package-manager-specific function. Currently, the only way I see to avoid this is to also introduce a `YarnDependencyHandler` which inherits from `NpmDependencyHandler`. But I'm not sure yet whether that's much nicer.